lib/_http_agent.js

```diff
@@ -342,10 +341,7 @@ Agent.prototype.destroy = function destroy() {
 function handleSocketCreation(request, informRequest) {
   return function handleSocketCreation_Inner(err, socket) {
     if (err) {
-      const asyncId = (socket && socket._handle && socket._handle.getAsyncId) ?
```

Comment:
After checking all the call sites, it's not possible to receive a `socket` here if `err` is truthy.